### ISSUE TITLE
Basic text file import issue

### ISSUE THREAD

**Issue by RS125man** [oric-software/OricExplorer#6]
I created a basic program (33 lines of code) as a text file in windows notepad and then used Oric Explorer V2.35 to import it as a ".tap" file. After this was completed, I loaded said tap file into Oricutron to test it. The program appeared to execute correctly in Oricutron and then returned to the curser in the default text mode as desired. However, when I then "listed" the program (to check/modify it further) the listing displayed extra lines of unexpected additional "random" characters at the end.  If the program was modified using "EDIT" to remove the extra unwanted characters, the text editor/Oric would crash and needed a system RESET. As this is appears to be some sort of data corruption and very unusual, I manually typed the very same program directly into the Oricutron application and then into a real Oric. Both worked correctly without any errors whatsoever when the program was executed/re-listed or edited.
My feeling is that there is some sort of data corruption at the point where the basic program (as text) is imported from the text file and then translated into the tap file. My request is that this feature is checked/investigated further.
I have attached the original text file that I used, to hopefully help identify the issue.
[Circles.txt](https://github.com/oric-software/OricExplorer/files/5870276/Circles.txt)

**Comment by dipisoft** [comment]
Hi,

Sorry for the delay of this answer but I didn't received notification of this issue.

I've just tested import of your "circles.txt" file with the new version of OricExplorer (v2.4) that I've just released and don't encounter any problem.

Can you retry with that new version when my pull request will be accepted?

**Comment by RS125man** [comment]
Hi,
I have retried the importing of my basic program (via a text file) using the latest version of Oric Explorer V2.4.1.0 and exactly the same issue appears to be present with this function. Please see the attached jpeg file that shows the characters that have been spuriously added at the end of the listing, once the program has been run and then re-listed.
![Circles_tap](https://user-images.githubusercontent.com/57588431/113490299-12cdac80-94c1-11eb-9a74-24c9c5467faf.jpg)

**Comment by dipisoft** [comment]
Hello,

I have also just repeated the same test several times with your circles.txt file and I arrive at the same result: no problem encountered.

So I imagine it's probably a matter of context. Which version of Windows? 32 or 64 bit version? Which version of Oricutron? As far as I'm concerned, I tested with v1.2 (from 11/01/2014) and "WIP rev: 8052493" (from 10/21/2016) on a Windows 10 pro x64 release 2004.

It remains to be seen whether the problem is indeed due to the generation of the .TAP file by OricExplorer (on your machine). This can also be attributed to treatment with Oricutron.

I therefore suggest that you test on your machine the .TAP file (joined in .zip) created on mine. For my part, I can test your .TAP file if you send it to me.

Cordially.

[circles.zip](https://github.com/oric-software/OricExplorer/files/6257073/circles.zip)

**Comment by dipisoft** [comment]
Ah there it is! I just noticed the anomaly after launching the program in basic.

Because when you import the file by unchecking the "Auto run" box and load the .TAP file into Oricutron (without launching the program), the listing is "normal". It is only after doing the RUN that the problem appears.

So I tested a different method to generate the .TAP: by pasting the listing in Oricutron then by saving it with a CSAVE. This allowed me to compare this file with the one generated by the OricExplorer import. And indeed, the two files are different.

It remains for me to understand where the difference comes from and to correct that.

To be continued...
